pkg/backup/backup.go

```diff
@@ -173,9 +173,30 @@ func getResourceHooks(hookSpecs []api.BackupResourceHookSpec, discoveryHelper di
 	return resourceHooks, nil
 }
 
+func getBackupValidationErrors(backup *api.Backup) []string {
```

Comment:
Make this a method on `kubernetesBackupper` and you can do this to move the remaining check from the controller to here:
```go
if kb.snapshotService == nil && boolptr.IsSetToTrue(backup.Spec.SnapshotVolumes) {
 // ...
}
```